flake.nix

```diff
@@ -35,11 +36,56 @@
               poetryOverrides
             ];
           };
+          spellcheck = pkgs.writeShellApplication {
+            name = "spellcheck";
+            runtimeInputs = with pkgs; [
+              moreutils
+              ncurses6
+              (hunspellWithDicts (with hunspellDicts; [
+                en-gb-ise
+                en-au
+                en-ca
+                en-us
+              ]))
+            ];
+            text = ''
+              mapfile -t files < <(find . -name '*.md')
+              exceptionsFile=./spelling-exceptions.dic
+              newExceptionsFile=./new-spelling-exceptions.dic
+              gbFile=gb.dic
+              otherFile=other.dic
+              hunspell -d en_GB -l "''${files[@]}" | sed 's/.*: //' | sort | uniq > $gbFile
+              hunspell -d en_US,en_AU,en_CA -l "''${files[@]}" | sed 's/.*: //' | sort | uniq > $otherFile
+              comm -1 -2 $gbFile $otherFile > $newExceptionsFile
+              mapfile -t newExceptions < <(comm -1 -3 $exceptionsFile $newExceptionsFile)
+              cat $exceptionsFile $newExceptionsFile | sort | uniq | sponge $exceptionsFile
+              rm $newExceptionsFile
+              if [ "''${#newExceptions[@]}" -gt 0 ]; then
+                echo New exceptions
+                tput setaf 3
+                IFS=$'\n'; echo "''${newExceptions[*]/#/ - }"
+                tput sgr 0
+                echo
+              fi
+              mapfile -t errors < <(comm -2 -3 $gbFile $exceptionsFile)
+              if [ "''${#errors[@]}" -gt 0 ]; then
+                echo Spellcheck errors
+                tput setaf 1
+                IFS=$'\n'; echo "''${errors[*]/#/ - }"
+                tput sgr 0
+                exit 1
+              fi
+            '';
+          };
           nix-dev-html = pkgs.stdenv.mkDerivation {
             name = "nix-dev";
             src = self;
-            buildInputs = [ packages.nix-dev-pyenv ];
+            buildInputs = [
+              packages.spellcheck
+              packages.nix-dev-pyenv
+            ];
             buildPhase = ''
+              spellcheck
```

Comment:
@Mic92 should be part of CI and output a warning instead of hard failure, in case of false positives. May be too annoying otherwise. GitHub has docs on which format the output has to have to be recognized as a warning.

https://github.com/marketplace/actions/check-spelling

Reply:
Excellent feedback, thank you.